### PR TITLE
[WIP] Convert documentation to yardoc

### PR DESCRIPTION
**Pull request by ekohl**
This isn't perfect since the examples aren't using @example but it's a step in the right direction

### CODE REVIEW

File: manifests/package/redhat.pp [hunk 1]
```diff
@@ -1,18 +1,4 @@
-# Class: nginx::package::redhat
-#
 # This module manages NGINX package installation on RedHat based systems
```
Comment (alexjfisher) [inline manifests/package/redhat.pp]: 'module'?

File: manifests/init.pp [hunk 2]
```diff
@@ -1,29 +1,136 @@
...
+# @param nginx_locations
+# @param nginx_locations_defaults
+# @param nginx_mailhosts
+# @param nginx_mailhosts_defaults
+# @param nginx_streamhosts
+# @param nginx_upstreams
+# @param nginx_servers
+# @param nginx_servers_defaults
+# @param purge_passenger_repo
+# @param add_listen_directive
```
Comment (alexjfisher) [inline manifests/init.pp]: Are these just place holders for people to fill in?  ie if you left this out, wouldn't puppet-strings generate the same content from the parameter list below?

File: manifests/config.pp [hunk 3]
```diff
@@ -1,18 +1,4 @@
...
-#
-# There are no default parameters for this class.
-#
-# Actions:
-#
-# Requires:
-#
-# Sample Usage:
-#
-# This class file is not called directly
```
Comment (alexjfisher) [inline manifests/config.pp]: `@api private`?

File: manifests/config.pp [hunk 4]
```diff
@@ -1,18 +1,4 @@
-# Class: nginx::config
-#
 # This module manages NGINX bootstrap and configuration
```
Comment (alexjfisher) [inline manifests/config.pp]: `# @summary This module manages NGINX bootstrap and configuration`?

**Comment by ekohl** [comment]
It looks like I missed some parts of the docs in the conversion.

**Comment by bastelfreak** [comment]
@ekohl ping

**Pull request closed without merging**